### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by valeriaoleynik**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
```
Comment (agniya-i) [inline src/transformStateWithClones.js:10]:
```suggestion
  const result = [];
```
use more meaningful variable names

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,30 @@
...
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
```
Comment (agniya-i) [inline src/transformStateWithClones.js:27]: add default case

**Review by agniya-i — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = Object.assign({}, state);
+  const caseDefault = 'An error because of an unknown type';
+
+  for (const action of actions) {
+    switch (action.type) {
```
Comment (polosanya) [inline src/transformStateWithClones.js:15]: You can also make a destructuring to get all the needed props of action (type, extraData, keysToRemove)

**Review by polosanya — APPROVED** [review]
Nice!

**Pull request closed without merging**